pkg/router/controller/unique_host.go

```diff
@@ -97,14 +116,17 @@ func (p *UniqueHost) HandleRoute(eventType watch.EventType, route *routeapi.Rout
 				if old[i].Spec.Path == route.Spec.Path {
 					if old[i].CreationTimestamp.Before(route.CreationTimestamp) {
 						glog.V(4).Infof("Route %s cannot take %s from %s", routeName, host, routeNameKey(oldest))
-						return fmt.Errorf("route %s holds %s and is older than %s", routeNameKey(oldest), host, key)
+						err := fmt.Errorf("route %s already exposes %s and is older", oldest.Name, host)
```

Comment:
Was it intentional to use 'oldest' here instead of 'old[i]'?  Also, would it make sense to include the path in the error message?